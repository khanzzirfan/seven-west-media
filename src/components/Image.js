import React from 'react';
import styled from 'styled-components';

const margin = 9;
const size = 38;

const Image = styled.img`
 margin-top:${margin}px;
 margin-left:${margin}px;
 width: ${size}px;
 height: ${size}px;
`;

export default Image;