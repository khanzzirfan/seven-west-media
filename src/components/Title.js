import React from 'react';
import styled from 'styled-components';

const marginLeft = 10;
const fontSize = 20;

const Title = styled.h2`
    color: white;
    margin: 0 0 0 ${marginLeft}px;
    font-size: ${fontSize}px;
    font-family: Roboto, sans-serif;
    display: flex;
    justify-content: center;
    flex-direction: column;
    text-align: center;
`;

export default Title;