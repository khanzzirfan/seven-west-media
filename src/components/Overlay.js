import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Image from './Image';
import Title from './Title';

const overlayViewHeight = 51;
const OverlayView = styled.div`
    display:flex;
    align-self: flex-end;
    background: rgba(0, 0, 0, 0.6);
    width: 100%;
    height: ${overlayViewHeight}px;
`;

const Overlay = ({ iconAlt, iconImage, ImageElement, TitleElement, title }) => (
    <OverlayView>
        <ImageElement src={iconImage} alt={iconAlt} />
        <TitleElement>{title}</TitleElement>
    </OverlayView>
);

Overlay.defaultProps = {
    iconAlt: '',
    iconImage: null,
    ImageElement: Image,
    TitleElement: Title,
    title: '',
};

Overlay.propTypes = {
    iconAlt: PropTypes.string,
    iconImage: PropTypes.node,
    ImageElement: PropTypes.func,
    TitleElement: PropTypes.func,
    title: PropTypes.string,
};

export default Overlay;