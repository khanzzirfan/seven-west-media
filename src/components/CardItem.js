import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Overlay from './Overlay';

const cardItemWidth = 320;
const cardItemHeight = 178;

const CardItemView = styled.div`
    display: flex;
    background-image: url(${props => props.backgroundImage});
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;
    width: ${cardItemWidth}px;
    height: ${cardItemHeight}px;
    margin: 10px 0 0 10px;
    position: relative;
    float: left;

    @media (max-width: 320px) {
        width: 100%;
    }
`;

const CardItem = ({ backgroundImage, iconImage, iconAlt, ImageElement, TitleElement, title }) => (
    <CardItemView backgroundImage={backgroundImage}>
        <Overlay iconAlt={iconAlt}
            iconImage={iconImage}
            ImageElement={ImageElement}
            TitleElement={TitleElement}
            title={title} />
    </CardItemView>
)

CardItem.propTypes = {
    backgroundImage: PropTypes.node,
    iconAlt: PropTypes.string,
    iconImage: PropTypes.node,
    ImageElement: PropTypes.func,
    TitleElement: PropTypes.func,
    title: PropTypes.string,
};


export default CardItem;