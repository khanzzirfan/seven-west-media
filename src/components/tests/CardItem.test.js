import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

import CardItem from '../CardItem';

import Overlay from '../Overlay';
import Image from '../Image';
import Title from '../Title';
import logoImage from '../../assets/logo.png';
import backgroundImage from '../../assets/tile.jpg';

describe('components -> CardItem component ', () => {
    let instance;
    let testRenderer;

    beforeAll(() => {
        testRenderer = TestRenderer.create(
            <CardItem
                backgroundImage={backgroundImage}
                iconAlt={"media alt"}
                iconImage={logoImage}
                ImageElement={Image}
                TitleElement={Title}
                title={"Home and Away"}

            />
        );
        instance = testRenderer.root;
    });

    it('should render correctly and match to snapshot', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("should pass props as background image to card item", () => {
        expect(instance.props.backgroundImage).toBe(backgroundImage);
    });

    it("should render overlay component as a part of card item", () => {
        const iconAlt = "media alt";
        const title = "Home and Away";
        expect(instance.findByType(Overlay).props.iconAlt).toBe(iconAlt);
        expect(instance.findByType(Overlay).props.title).toBe(title);
    });

});