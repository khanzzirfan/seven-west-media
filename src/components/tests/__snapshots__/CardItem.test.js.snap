// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components -> CardItem  should render correctly and match to snapshot 1`] = `
.c2 {
  margin-top: 9px;
  margin-left: 9px;
  width: 38px;
  height: 38px;
}

.c3 {
  color: white;
  margin: 0 0 0 10px;
  font-size: 20px;
  font-family: Roboto,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background: rgba(0,0,0,0.6);
  width: 100%;
  height: 51px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-image: url(tile.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  width: 320px;
  height: 178px;
  margin: 10px 0 0 10px;
  position: relative;
  float: left;
}

@media (max-width:320px) {
  .c0 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      alt="media alt"
      className="c2"
      src="logo.png"
    />
    <h2
      className="c3"
    >
      Home and Away
    </h2>
  </div>
</div>
`;
